### DISCUSSION TITLE
On new setup, clicking the button to generate the env seed didn't work.

### DISCUSSION THREAD

**Discussion by accessiblepixel** [JulianPrieber/littlelink-custom#87] [Issues]
When I setup my new copy of little-link custom, I had to set that random salt value in the env file manually, since the version to click within the software wouldn't work for me.

I couldn't trace any reason why it wasn't working (there was no output in PHP's error_log) or any errors in apache2's log so I couldn't ultimately figure out what the root cause of the issue was, but I thought I'd better report that I'd had issue with it since it might confuse new (er) users, which from what I can tell is one of the design goals of this version :)

It was fixed by manually generating the salt and adding it to the .env file.

Kind regards,
Jessica

**Comment by lastsamurai26** [comment]
Thank you for the message. With version 2.6.9 should give a suitable error 
message if the software can not generate the key. The software automatically 
calls the artisan key:generate command if appkey is not defined in the .env file.

This is the normal error messege, There is actually no more button
![chrome_ya6LxYqjqJ](https://user-images.githubusercontent.com/3257844/173684059-97f49cc9-593a-41ad-84b5-f05fc74d08b6.png)


Which version did you installed ?

best regards, 
frank

**Comment by lastsamurai26** [comment]
Hey, 

I have tested it again. 
Installed version 2.6.9. Can you please check if the .env file has 644
permissions and if the webserver can access the files.

**Comment by accessiblepixel** [comment]
Hi!

I checked permissions and they were all 644 for the files and 755 for the directories, which is what the documentation said, so it should have had full permissions to write to the files.

I thought it didn't make much sense so I looked for a manual way to set that key and then everything magically worked after that... so it must have been something with the call to write to that file, but I can't seen to see anywhere that was telling me it had failed or was denied.

As I remember when I set it up, it appeared to process the command just fine, but wouldn't actually do it (it didn't say permission denied as above).

I'm running it on PHP 7.4 (with Apache hooked in with LSAPI on CloudLinux).

Is there anything else I could check to try and help you find out?

Kind regards,
Jessica

**Comment by JulianPrieber** [comment]
**Hi,** 
_don't mind me sliding in._

In case you didn't do this already, did you verify the file ownership? In most cases, you can confirm this with: 
`ls -l /path/to/littlelink-custom`
_and or_
`stat /path/to/littlelink-custom`

Assuming `www-data` (it could be something else) is your web server user, you can set file ownership with:
`sudo chown -R www-data:www-data /path/to/littlelink-custom`

There might be some useful information in the Laravel log, located in `/path/to/littlelink-custom/storage/logs/laravel.log`.

Other than that, you can find a list of required PHP and modules on [here](https://littlelink-custom.com/docs/d/installation-requirements/).

All the best,
Julian

**Comment by JulianPrieber** [comment]
Migrated to discussion.
